File: AMF5/src/test/java/javaPlatform/JSONSchemaToDialectTest.java
package javaPlatform;

import amf.aml.client.platform.AMLBaseUnitClient;
import amf.aml.client.platform.AMLConfiguration;
import amf.aml.client.platform.AMLDialectInstanceResult;
import amf.aml.client.platform.AMLDialectResult;
import amf.aml.client.platform.model.document.Dialect;
import amf.aml.client.platform.model.document.DialectInstance;
import amf.shapes.client.platform.config.SemanticJsonSchemaConfiguration;
import org.junit.Test;

import java.io.IOException;
import java.util.concurrent.ExecutionException;

import static javaPlatform.StringEquals.assertIgnoringWhitespace;
import static org.hamcrest.CoreMatchers.instanceOf;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.junit.Assert.assertTrue;

public class JSONSchemaToDialectTest implements FileReader {

    final String inputJsonSchema = "src/test/resources/examples/json-schema-dialect.json";
    final String outputAmlDialect = "/examples/json-schema-dialect.yaml";
    final String jsonSchemaInstance = "src/test/resources/examples/json-instance.json";

    @Test
    public void convertJSONSchemaToDialect() throws IOException, ExecutionException, InterruptedException {
        AMLBaseUnitClient client = SemanticJsonSchemaConfiguration.predefined().baseUnitClient();
        AMLDialectResult parseResult = client.parseDialect("file://" + inputJsonSchema).get();

        assertTrue(parseResult.conforms());
        assertThat(parseResult.baseUnit(), instanceOf(Dialect.class));

        Dialect generatedDialect = parseResult.dialect();
        String readDialect = readResource(outputAmlDialect);
        String renderDialect = client.render(generatedDialect);

        assertIgnoringWhitespace(readDialect, renderDialect);
    }

    @Test
    public void validateJSONInstanceWithGeneratedDialect() throws ExecutionException, InterruptedException {
        AMLBaseUnitClient semanticJsonSchemaClient = SemanticJsonSchemaConfiguration.predefined().baseUnitClient();
        AMLDialectResult dialectResult = semanticJsonSchemaClient.parseDialect("file://" + inputJsonSchema).get();

        assertTrue(dialectResult.conforms());
        assertThat(dialectResult.baseUnit(), instanceOf(Dialect.class));

        AMLBaseUnitClient amlClient = AMLConfiguration.predefined().withDialect(dialectResult.dialect()).baseUnitClient();
        AMLDialectInstanceResult instanceResult = amlClient.parseDialectInstance("file://" + jsonSchemaInstance).get();

        assertTrue(instanceResult.conforms());
        assertThat(instanceResult.baseUnit(), instanceOf(DialectInstance.class));
    }

}

File: AMF5/src/test/scala/scalaPlatform/JSONSchemaToDialectTest.scala
package scalaPlatform

import amf.aml.client.scala.AMLConfiguration
import amf.aml.client.scala.model.document.{Dialect, DialectInstance}
import amf.shapes.client.scala.config.SemanticJsonSchemaConfiguration
import org.scalatest.flatspec.AsyncFlatSpec
import org.scalatest.matchers.should

import scala.concurrent.Future

class JSONSchemaToDialectTest extends AsyncFlatSpec with should.Matchers with FileUtils {

  val inputJsonSchema = "src/test/resources/examples/json-schema-dialect.json"
  val outputAmlDialect = "src/test/resources/examples/json-schema-dialect.yaml"
  val jsonSchemaInstance = "src/test/resources/examples/json-instance.json"

  it should "Convert JSON Schema to Dialect" in {
    val client = SemanticJsonSchemaConfiguration.predefined().baseUnitClient()
    client.parseDialect(addPrefix(inputJsonSchema)) map { parseResult =>
      parseResult.conforms shouldBe true
      parseResult.baseUnit.isInstanceOf[Dialect] shouldBe true

      val generatedDialect = parseResult.dialect
      val readDialect = getStrFromFile(outputAmlDialect)
      val renderDialect = client.render(generatedDialect)

      renderDialect.trim shouldEqual readDialect.trim
    }
  }

  it should "Validate JSON instance with generated Dialect" in {
    val semanticJsonSchemaClient = SemanticJsonSchemaConfiguration.predefined().baseUnitClient()
    for {
      dialectResult <- semanticJsonSchemaClient.parseDialect(addPrefix(inputJsonSchema))
      amlClient <-
        Future(AMLConfiguration.predefined().withDialect(dialectResult.dialect).baseUnitClient())
      instanceResult <- amlClient.parseDialectInstance(addPrefix(jsonSchemaInstance))
    } yield {
      dialectResult.conforms shouldBe true
      dialectResult.baseUnit.isInstanceOf[Dialect] shouldBe true
      instanceResult.conforms shouldBe true
      instanceResult.baseUnit.isInstanceOf[DialectInstance] shouldBe true
    }
  }

}
